### PR TITLE
DAOS-6340 utils: daos container check

### PR DESCRIPTION
**Pull request by Nasf-Fan**
New container based daos untils command option "daos cont check":
it list all the objects in the given container, for each of them,
verify its data consistency.

Currently, EC objects are skipped. We will support to verify data
consistency for EC object sometime later.

Signed-off-by: Fan Yong <fan.yong@intel.com>

### CODE REVIEW

File: src/utils/daos_hdlr.c [hunk 1]
```diff
@@ -655,6 +656,77 @@ pool_query_hdlr(struct cmd_args_s *ap)
...
+				"object IDs enumeration failed: "DF_RC"\n",
+				DP_RC(rc));
+			D_GOTO(out_close, rc);
+		}
+
+		for (i = 0; i < oids_nr; i++) {
+			rc = daos_obj_verify(ap->cont, oids[i], ap->epc);
+			if (rc == -DER_NOSYS) {
+				/* XXX: NOT support to verif EC object yet. */
+				D_PRINT("spik check for object: "DF_OID"\n",
```
Comment (mchaarawi) [inline src/utils/daos_hdlr.c:697]: typo: "spik" ?

File: src/utils/daos_hdlr.c [hunk 2]
```diff
@@ -655,6 +656,77 @@ pool_query_hdlr(struct cmd_args_s *ap)
...
+			}
+
+			if (rc < 0) {
+				fprintf(stderr,
+					"check object "DF_OID" failed: "
+					DF_RC"\n", DP_OID(oids[i]), DP_RC(rc));
+				D_GOTO(out_close, rc);
+			}
+
+			D_PRINT("checked object: "DF_OID"\n", DP_OID(oids[i]));
```
Comment (mchaarawi) [inline src/utils/daos_hdlr.c:718]: do we want to print this for every object? we could have millions..

File: src/utils/daos.c [hunk 3]
```diff
@@ -1328,6 +1333,7 @@ do { \
 	"	  delete-acl       delete an entry from a container's ACL\n" \
 	"	  set-owner        change the user and/or group that own a container\n" \
 	"	  stat             get container statistics\n" \
+	"	  check            check objects consistency in container\n" \
```
Comment (mchaarawi) [inline src/utils/daos.c:1336]: check is very general, and could apply to objects, snapshots, etc.
how about check-obj or verify-obj?

**Review by mchaarawi — COMMENTED** [review]
should also update:
doc/man/man8/daos.8

File: src/utils/daos_hdlr.c (hunk 2 above)
Comment (liuxuezhao) [reply src/utils/daos_hdlr.c:718]: using "D_INFO" probably is fine.

**Review by liuxuezhao — DISMISSED** [review]

File: src/utils/daos_hdlr.c (hunk 1 above)
Comment (Nasf-Fan) [reply src/utils/daos_hdlr.c:697]: Sorry, it should be "skip".

File: src/utils/daos.c (hunk 3 above)
Comment (Nasf-Fan) [reply src/utils/daos.c:1336]: Originally, I used "verify-objects", but according to the description in DAOS-6340, my understand is that Johann wants something like DAOS filesystem checker. If that is correct, "check" may be more consistent with others name, such as "e2fsck" or "lfsck".
@johannlombardi , what is your suggestion?

File: src/utils/daos_hdlr.c (hunk 2 above)
Comment (Nasf-Fan) [reply src/utils/daos_hdlr.c:718]: Maybe we only need to log the inconsistent or skipped objects?

File: src/utils/daos.c (hunk 3 above)
Comment (johannlombardi) [reply src/utils/daos.c:1336]: The intent is to verify more and more under "container check", so i would stick to this.

File: src/utils/daos_hdlr.c (hunk 1 above)
Comment (johannlombardi) [reply src/utils/daos_hdlr.c:697]: Might be better to count the number of objects that were skipped and effectively verified and then print a summary at the end.

File: src/utils/daos_hdlr.c (hunk 2 above)
Comment (johannlombardi) [reply src/utils/daos_hdlr.c:718]: I agree with Mohamad. Let's just print a summary at the end.

**Review by johannlombardi — CHANGES_REQUESTED** [review]
Thanks for patch. A few improvement on the logging is required before it can land.

**Review by johannlombardi — DISMISSED** [review]

**Review by mchaarawi — APPROVED** [review]

**Review by liuxuezhao — DISMISSED** [review]

**Comment by Nasf-Fan** [comment]
CI test timeout after "daos_test -z". It is DAOS-6497. I do not think it is related with daos utils changes.
So can we land this patch? @johannlombardi @daos-stack/daos-gatekeeper

**Comment by Nasf-Fan** [comment]
Hit DAOS-6497 again. But I do not "daos_test -z" timeout is related with daos utils changes.

**Comment by Nasf-Fan** [comment]
Update man page for new "daos cont check".

**Review by johannlombardi — APPROVED** [review]

**Review by rpadma2 — APPROVED** [review]
LGTM

**Comment by Nasf-Fan** [comment]
daos_test -z failed for DAOS-6507.

**Pull request merged**